### DISCUSSION TITLE
[Enhancement] Permissions

### DISCUSSION THREAD

**Discussion by shehi** [discord-php/DiscordPHP#1052] [Ideas]
I think it makes sense to convert all Permission keys https://github.com/discord-php/DiscordPHP/blob/master/src/Discord/Parts/Permissions/Permission.php#L35  to constants. Also it would make more sense to rename these Permission keys to their Discord-documented counterparts snake-cased (e.g. `use_slash_commands` should become `use_application_commands` etc) https://discord.com/developers/docs/topics/permissions . I can send in PR. Thoughts?

**Comment by davidcole1340** [comment]
Not 100% sure what you mean about converting all keys to constants, could you show an example? Since you can set a permission via `$permission->manage_channels = true` ideally you don't have to work with the raw value.

And yeah, ideally we match the Discord documentation, however I am weary of making this a breaking change. We should be able to add the correct name alongside the old name to keep backwards compatibility, just leaving a note of which one is deprecated. 

e.g.
```
'use_slash_commands' => (1 << 31),
'use_application_commands' => (1 << 31),
```

**Comment by shehi** [comment]
The problem I see lies in those string literal indices: `use_slash_commands` etc. I believe they should be made constants. And instead of `$permission->manage_channels` which contains these literal-strings-gone-property, just use something like `$permission->allows(Permission::USE_APPLICATION_COMMANDS)` or similar, which decouples the remainder of the codebase from these hardcoded literals a bit. This way, it will also be possible to deprecate constants via PHPDocs in future if need be. 

Nothing critical here of course, just good old refactoring.
